azure-servicemanagement-legacy/tests/test_legacy_mgmt_misc.py

```diff
@@ -2166,7 +2166,7 @@ def test_list_os_images_public(self):
         self.assertIsNotNone(image.is_premium)
         self.assertIsNotNone(image.icon_uri)
         self.assertIsNotNone(image.privacy_uri)
-        self.assertGreaterEqual(len(image.recommended_vm_size), 0)
+        self.assertIsNotNone(image.recommended_vm_size)
```

Comment:
Does that actually catch cases where the field is unpopulated? I would have assumed that `len(None)` would have blown up in the previous version of the test code.

For example, why `assertIsNotNone` instead of `assertGreater`  than zero list is done for the `publisher_name` which is also a string field?